Fyne does not compile on FreeBSD

Hello,

I am having problems getting fyne to work on FreeBSD. I do not know if this is a bug or if I have forgotten something and ask you here for help.

I have set up a fresh FreeBSD and installed the dependencies as described on https://developer.fyne.io/started/ with `sudo pkg install go gcc xorg glfw pkgconf`.

After that I proceeded as follows, where the installation aborted with an error message:
```
$ go get fyne.io/fyne/v2
go: downloading fyne.io/fyne/v2 v2.0.2
go: downloading fyne.io/fyne v1.4.3

$ go get fyne.io/fyne/v2/cmd/fyne_demo
go: downloading github.com/srwiley/oksvg v0.0.0-20200311192757-870daf9aa564
go: downloading github.com/srwiley/rasterx v0.0.0-20200120212402-85cb7272f5e9
go: downloading golang.org/x/image v0.0.0-20200430140353-33d19683fad8
go: downloading github.com/goki/freetype v0.0.0-20181231101311-fa8a33aabaff
go: downloading github.com/fsnotify/fsnotify v1.4.9
go: downloading github.com/fyne-io/mobile v0.1.3-0.20210318200029-09e9c4e13a8f
go: downloading github.com/godbus/dbus/v5 v5.0.3
go: downloading golang.org/x/sys v0.0.0-20200720211630-cb9d2d5c5666
go: downloading github.com/fredbi/uri v0.0.0-20181227131451-3dcfdacbaaf3
go: downloading github.com/stretchr/testify v1.5.1
go: downloading github.com/go-gl/glfw/v3.3/glfw v0.0.0-20210311203641-62640a716d48
go: downloading github.com/go-gl/gl v0.0.0-20190320180904-bf2b1f2f34d7
go: downloading gopkg.in/yaml.v2 v2.2.8
go: downloading github.com/davecgh/go-spew v1.1.1
go: downloading github.com/pmezard/go-difflib v1.0.0
go: downloading golang.org/x/net v0.0.0-20200324143707-d3edc9973b7e
go: downloading golang.org/x/text v0.3.2
# github.com/godbus/dbus/v5
In file included from go/pkg/mod/github.com/godbus/dbus/v5@v5.0.3/transport_unixcred_freebsd.go:13:
In file included from /usr/include/sys/ucred.h:39:
/usr/include/sys/_lock.h:38:2: error: unknown type name 'u_int'
        u_int   lo_flags;
        ^
/usr/include/sys/_lock.h:39:2: error: unknown type name 'u_int'
        u_int   lo_data;                /* General class specific data. */
        ^
In file included from go/pkg/mod/github.com/godbus/dbus/v5@v5.0.3/transport_unixcred_freebsd.go:13:
In file included from /usr/include/sys/ucred.h:40:
/usr/include/sys/_mutex.h:49:11: error: unknown type name 'uintptr_t'
        volatile uintptr_t      mtx_lock;       /* Owner and flags. */
                 ^
/usr/include/sys/_mutex.h:63:11: error: unknown type name 'uintptr_t'
        volatile uintptr_t      mtx_lock;       /* Owner and flags. */
                 ^
4 errors generated.
# github.com/go-gl/glfw/v3.3/glfw
In file included from go/pkg/mod/github.com/go-gl/glfw/v3.3/glfw@v0.0.0-20210311203641-62640a716d48/native_linbsd.go:10:
In file included from go/pkg/mod/github.com/go-gl/glfw/v3.3/glfw@v0.0.0-20210311203641-62640a716d48/glfw/include/GLFW/glfw3native.h:114:
In file included from /usr/local/include/GL/glx.h:32:
/usr/local/include/GL/gl.h:56:11: warning: 'GLAPIENTRY' macro redefined [-Wmacro-redefined]
go/pkg/mod/github.com/go-gl/glfw/v3.3/glfw@v0.0.0-20210311203641-62640a716d48/glfw/include/GLFW/glfw3.h:5894:10: note: previous definition is here
```

It looks like the problem is not with fyne, but with godbus / dbus.

```
$ go get github.com/godbus/dbus
go: downloading github.com/godbus/dbus v4.1.0+incompatible
# github.com/godbus/dbus
In file included from go/pkg/mod/github.com/godbus/dbus@v4.1.0+incompatible/transport_unixcred_freebsd.go:13:
In file included from /usr/include/sys/ucred.h:39:
/usr/include/sys/_lock.h:38:2: error: unknown type name 'u_int'
        u_int   lo_flags;
        ^
/usr/include/sys/_lock.h:39:2: error: unknown type name 'u_int'
        u_int   lo_data;                /* General class specific data. */
        ^
In file included from go/pkg/mod/github.com/godbus/dbus@v4.1.0+incompatible/transport_unixcred_freebsd.go:13:
In file included from /usr/include/sys/ucred.h:40:
/usr/include/sys/_mutex.h:49:11: error: unknown type name 'uintptr_t'
        volatile uintptr_t      mtx_lock;       /* Owner and flags. */
                 ^
/usr/include/sys/_mutex.h:63:11: error: unknown type name 'uintptr_t'
        volatile uintptr_t      mtx_lock;       /* Owner and flags. */
                 ^
4 errors generated.
```
https://github.com/godbus/dbus/issues/228
https://github.com/godbus/dbus/issues/225

I am trying to figure out what has changed since 12.2. As soon as I know more, I'll let you know.

Thank you for your great and fast support. :-)

--- a/go.mod
+++ b/go.mod
@@ -10,7 +10,7 @@ require (
 	github.com/fyne-io/mobile v0.1.3-0.20210318200029-09e9c4e13a8f
 	github.com/go-gl/gl v0.0.0-20190320180904-bf2b1f2f34d7
 	github.com/go-gl/glfw/v3.3/glfw v0.0.0-20210311203641-62640a716d48
-	github.com/godbus/dbus/v5 v5.0.3
+	github.com/godbus/dbus/v5 v5.0.4
 	github.com/goki/freetype v0.0.0-20181231101311-fa8a33aabaff
 	github.com/jackmordaunt/icns v0.0.0-20181231085925-4f16af745526
 	github.com/josephspurrier/goversioninfo v0.0.0-20200309025242-14b0ab84c6ca

